ikarus/finiteelements/feresulttypes.hh

```diff
@@ -0,0 +1,156 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#pragma once
+
+#include <type_traits>
+
+#include <ikarus/utils/tensorutils.hh>
+
+/**
+ *
+ * \ingroup FEParameterTags
+ * \brief A strongly typed enum class representing the type of the result request
+ */
+
+namespace Ikarus {
+namespace Util {
+
+  struct VectorizeWithVoigt
```

Comment:
Maybe we put these in an Impl namespace since they are an implementation detail.